src/shogun/machine/GLM.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+#include <shogun/features/DotFeatures.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/SGDMinimizer.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+// #include <utility>
+#include <cmath>
+
+using namespace shogun;
+
+GLM::GLM(GLM_DISTRIBUTION distr, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta): IterativeMachine<LinearMachine>()
+{
+	distribution=distr;
+	m_alpha=alpha;
+	m_lambda=lambda;
+	m_max_iterations=max_iterations;
+	m_tolerance=tolerance;
+	m_eta=eta;
+
+	m_learning_rate->set_const_learning_rate(2e-1);
+	m_learning_rate->set_const_learning_rate(learning_rate);
+	
+	m_cost_function->set_target(std::make_shared<GLM> (this));
+
+	m_penalty->set_l1_ratio(m_alpha);
+}
+
+std::shared_ptr<RegressionLabels> GLM::apply_regression(std::shared_ptr<Features> data)
+{
+	LinearMachine::set_features(std::static_pointer_cast<DotFeatures>(data));
+
+	if (!LinearMachine::features)
+		return std::make_shared<RegressionLabels>(SGVector<float64_t>());
+
+	int32_t num=LinearMachine::features->get_num_vectors();
+	ASSERT(num>0)
+	ASSERT(m_w.vlen==features->get_dim_feature_space())
+	SGVector<float64_t> out(num);
+	LinearMachine::features->dense_dot_range(out.vector, 0, num, NULL, m_w.vector, m_w.vlen, bias);
+	SGVector<float64_t> result = m_cost_function->non_linearity(out);
+	return std::make_shared<RegressionLabels>(result);
+}
+
+void GLM::init_model(std::shared_ptr<Features> data)
+{
+
+	LinearMachine::set_features(std::static_pointer_cast<DotFeatures>(data));
+	NormalDistribution<float64_t> normal_dist;
+	auto n_features = LinearMachine::features->get_dim_feature_space();
+	
+	if (m_compute_bias && bias == 0)
+		bias = 1 / (n_features + 1) * normal_dist(m_prng);
```

Comment:
@vigsterkr This is where I use `m_prng` from the `RandomMixin` class

Reply:
yeah but GLM is not implementing any Distribution interface...